docs/fundamentals/code-analysis/quality-rules/ca1844.md

```diff
@@ -0,0 +1,48 @@
+---
+title: "CA1844: Provide memory-based overrides of async methods when subclassing 'Stream' (code analysis)"
+description: "Learn about code analysis rule CA1844: Provide memory-based overrides of async methods when subclassing 'Stream'"
+ms.date: 05/18/2021
+ms.topic: reference
+f1_keywords:
+- CA1844
+- ProvideStreamMemoryBasedAsyncOverrides
+helpviewer_keywords:
+- ProvideStreamMemoryBasedAsyncOverrides
+- CA1844
+author: NewellClark
+dev_langs:
+- CSharp
+- VB
+---
+# CA1844: Provide memory-based overrides of async methods when subclassing 'Stream'
+
+| | Value |
+|-|-|
+| **Rule ID** |CA1844|
+| **Category** |[Performance](performance-warnings.md)|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+A type derived from <xref:System.IO.Stream> overrides <xref:System.IO.Stream.ReadAsync(System.Byte[],System.Int32,System.Int32,System.Threading.CancellationToken)> or <xref:System.IO.Stream.WriteAsync(System.Byte[],System.Int32,System.Int32,System.Threading.CancellationToken)> but does not override <xref:System.IO.Stream.ReadAsync(System.Memory{System.Byte}, System.Threading.CancellationToken)> or <xref:System.IO.Stream.WriteAsync(System.ReadOnlyMemory{System.Byte},System.Threading.CancellationToken)>, respectively.
```

Comment:
This is the only warning left for this article. If the build failed after fixing it, then it's failing for the other warnings unrelated to your PR (which you can ignore for now).

```suggestion
A type derived from <xref:System.IO.Stream> overrides <xref:System.IO.Stream.ReadAsync(System.Byte[],System.Int32,System.Int32,System.Threading.CancellationToken)> or <xref:System.IO.Stream.WriteAsync(System.Byte[],System.Int32,System.Int32,System.Threading.CancellationToken)> but does not override <xref:System.IO.Stream.ReadAsync(System.Memory{System.Byte},System.Threading.CancellationToken)> or <xref:System.IO.Stream.WriteAsync(System.ReadOnlyMemory{System.Byte},System.Threading.CancellationToken)>, respectively.
```